reference-implementation/lib/writable-stream.js

```diff
@@ -159,31 +151,74 @@ function WritableStreamAbort(stream, reason) {
     };
   });
 
+  if (state !== 'erroring') {
+    WritableStreamSetPendingError(stream, error);
+  }
+
   return promise;
 }
 
-function WritableStreamError(stream, error) {
-  stream._state = 'errored';
-  stream._storedError = error;
+function WritableStreamSetPendingError(stream, reason) {
```

Comment:
Maybe this should be renamed WritableStreamBecomeErroring or similar? Might be fine as-is.